### PR TITLE
Prompt fixes for non-interactive terminals

### PR DESCRIPTION
**Pull request by emosbaugh**
This pull request will change the following:

1. Disable all prompt timeouts for interactive terminals
1. Auto-enter confirm prompts with no timeout for non-interactive terminals
1. Fail for all prompts that require user input for non-interactive terminals
1. Disable the preflight failure prompts entirely until we are ready to add them back since timeout of 10 seconds only causes confusion and bug reports
1. Added properties `Kurl.IgnoreRemoteLoadImagesPrompt` and `Kurl.IgnoreRemoteUpgradePrompt` to bypass prompts for automation.

### CODE REVIEW

File: scripts/common/addon.sh [hunk 1]
```diff
@@ -144,9 +144,14 @@ function addon_outro() {
...
         fi
-        printf "Press enter to proceed\n"
-        prompt
 
+        if prompts_can_prompt ; then
+            echo "Press enter to proceed"
+            prompt
+        else
+            echo "TODO: what to do here"
+            # logWarn "Upgrade script must be run on all remote nodes"
```
Comment (emosbaugh) [inline scripts/common/addon.sh:153]: I'm not sure what to do in this case. There is no way to automate this currently unless it continues. Should there be a warning? Or should we add an additional flag?

File: scripts/common/preflights.sh [hunk 2]
```diff
@@ -194,7 +194,7 @@ checkFirewalld() {
     fi
 
     printf "${YELLOW}Continue with firewalld active? ${NC}"
-    if confirmY ; then
+    if confirmN ; then
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:197]: Switching to fail by default. Use BYPASS_FIREWALLD_WARNING.

File: scripts/common/preflights.sh [hunk 3]
```diff
@@ -238,7 +238,7 @@ checkUFW() {
     fi
 
     printf "${YELLOW}Continue with ufw active? ${NC}"
-    if confirmY ; then
+    if confirmN ; then
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:241]: Switching to fail by default. Use BYPASS_UFW_WARNING.

File: scripts/install.sh [hunk 4]
```diff
@@ -199,9 +199,10 @@ EOF
 
                 printf "${GREEN}    ${prefix}tasks.sh | sudo bash -s set-kubeconfig-server https://${currentLoadBalancerAddress}${NC}\n"
             fi
+
             printf "\n"
             printf "Continue? "
-            confirmY " "
+            confirmN
```
Comment (emosbaugh) [inline scripts/install.sh:205]: Switching to fail by default. Run load images task on other servers to bypass.

File: scripts/common/prompts.sh [hunk 5]
```diff
@@ -230,38 +235,26 @@ function prompt_airgap_preload_images() {
...
+        if ! prompts_can_prompt ; then
+            unattended_nodes_missing_images=1
+            continue
+        fi
+
         while true; do
             echo ""
             printf "Have images been loaded on node ${nodeName}? "
-            if confirmY " "; then
+            if confirmN ; then
```
Comment (emosbaugh) [inline scripts/common/prompts.sh:246]: Switching to fail by default. Run task on other servers to bypass.

File: scripts/common/prompts.sh [hunk 6]
```diff
@@ -230,38 +235,26 @@ function prompt_airgap_preload_images() {
...
             fi
         done
     done < <(kubectl get nodes --no-headers)
-}
 
-promptForPublicIp() {
-    if [ -n "$PUBLIC_ADDRESS" ]; then
-        return 0;
+    if [ "$unattended_nodes_missing_images" = "1" ] ; then
+        bail "Preloading images required"
```
Comment (emosbaugh) [inline scripts/common/prompts.sh:253]: Switching to fail by default. Run task on other servers to bypass.

**Comment by emosbaugh** [comment]
Example with non-interactive shell

```
[ethan@ethan-kurl-auto-2 ~]$ cat join.sh | sudo bash -s  kubeadm-token=n6w48g.35sqbv8g1syxk57g kubeadm-token-ca-hash=sha256:a5cfb26a9c0ee4c7ff8fe6648efaee5c783e1c3ee449c4198179df296196f065 kubernetes-version=1.19.10 docker-registry-ip=10.96.2.147 primary-host=10.128.0.90 | tee
Downloading package kurl-bin-utils-v2021.05.05-0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 81.7M  100 81.7M    0     0  40.0M      0  0:00:02  0:00:02 --:--:-- 40.0M
Firewalld is active, please press Y to disable (Y/n) Y
Automatically accepting prompt, shell is not interactive
Removed symlink /etc/systemd/system/multi-user.target.wants/firewalld.service.
Removed symlink /etc/systemd/system/dbus-org.fedoraproject.FirewallD1.service.

Kubernetes is incompatible with SELinux. Disable SELinux to continue? (Y/n) Y
Automatically accepting prompt, shell is not interactive
kubernetes.masterAddress required
[ethan@ethan-kurl-auto-2 ~]$
```

File: scripts/common/addon.sh (hunk 1 above)
Comment (AmberAlston) [reply scripts/common/addon.sh:153]: We want to be able to support full automation if desirable, so preferably a flag.

Comment (emosbaugh) [reply scripts/common/addon.sh]: Added properties Kurl.IgnoreRemoteLoadImagesPrompt and Kurl.IgnoreRemoteUpgradePrompt to bypass prompts for automation.

Comment (AmberAlston) [reply scripts/common/addon.sh]: Cool, thanks. Let's make sure we doc those

Comment (emosbaugh) [reply scripts/common/addon.sh]: https://github.com/replicatedhq/kurl.sh/pull/413

**Comment by emosbaugh** [comment]
Docs https://github.com/replicatedhq/kurl.sh/pull/413

**Review by areed — APPROVED** [review]

**Pull request merged**